Softeq.XToolkit.Common/Collections/ObservableRangeCollection.cs

```diff
@@ -211,22 +211,19 @@ public void InsertRange(IEnumerable<T> collection, int startIndex)
                 return;
             }
 
-            var changedItems = collection is List<T> ? (List<T>) collection : new List<T>(collection);
-            var index = Items.IndexOf(changedItems[0]);
-
-            for (var i = 0; i < changedItems.Count; i++)
+            var removedItems = new List<T>();
+            foreach (var item in collection)
             {
-                if (!Items.Remove(changedItems[i]))
+                if (Items.Remove(item))
                 {
-                    changedItems.RemoveAt(i); // Can't use a foreach because changedItems is intended to be (carefully) modified
-                    i--;
+                    removedItems.Add(item);
```

Comment:
Broken the previous comment.
Enumerable collection can be changed.

Reply:
As I understand, the logic was the following:
1. `collection` list is passed as a param of the RemoveRange method
2. `changedItems` list is initialized with the `collection` items
3. Then for each item of the `changedItems` list, we're trying to remove it from our ObservableRangeCollection, and if it's NOT removed, we're removing it from the `changedItems` list
4. `changedItems` list is changed while enumerating so that's why we couldn't use `foreach` here

Right now the logic has changed to initialize `changedItems` with empty collection and add item to it only if the item is removed from the ObservableRangeCollection. Also, we're enumerating through the original `collection` list, not through the `changedItems`.
As I understand, the problem that was mentioned in this comment is no longer actual

Reply:
Agree. Thanks for the detailed description!